share/toolchain-nxdk.cmake

```diff
@@ -45,4 +52,34 @@ set(_CMAKE_C_IPO_SUPPORTED_BY_CMAKE YES)
 set(_CMAKE_C_IPO_MAY_BE_SUPPORTED_BY_COMPILER YES)
 set(CMAKE_C_COMPILE_OPTIONS_IPO -flto)
 
+# TODO: CMAKE_MODULE_PATH is intended for project level than toolchain,
+#       find out what's wrong with CMAKE_FIND_ROOT_PATH as it is not working for some reason?
+set(CMAKE_MODULE_PATH "${NXDK_DIR}/share/cmake/Modules")
 set(PKG_CONFIG_EXECUTABLE "${NXDK_DIR}/bin/nxdk-pkg-config" CACHE STRING "Path to pkg-config")
+
+# Additional toolchain steps required for create xbe and xiso files.
+macro(add_executable)
+  # Perform normal call since there's no modification needed except for additional steps require
+  _add_executable(${ARGV})
+
+  # If author request generate iso file, then prepare the command here.
+  if(GEN_XISO)
+    set(XISO_OUTPUT "${NXDK_DIR}/tools/extract-xiso/build/extract-xiso" -c "$<TARGET_FILE_DIR:${ARGV0}>/bin" "$<TARGET_FILE_DIR:${ARGV0}>/${GEN_XISO}")
+  endif()
+
+  # If custom title is not given, use given executable name.
+  if(NOT XBE_TITLE)
```

Comment:
I believe this variable name needs a prefix to not clash with user variables in CMake?

There's probably some convention for this in CMake.

Reply:
It is only triggered when user call add_executable function. Though, I will check with forked NevolutionX repo with fixed CMake support to be sure it doesn't return back to user's CMakeList. It may also apply to `GEN_XISO` variable too.

Reply:
I can confirm XBE_TITLE is stored back into user's CMakeList, will need to make a fix for this.

concept fix:
```cmake
  if(NOT XBE_TITLE)
    set(<prefix>_XBE_TITLE ${ARGV0})
  else()
    set(<prefix>_XBE_TITLE ${XBE_TITLE})
  endif()
```
As for prefix, not sure if we should use NXDK or CMAKE prefix.

Reply:
> Could we also do an `unset` when we're done with it?

That's the plan. I must had forgot to mentioned that in my previous comment.